### PR TITLE
Forward various connection options to compute nodes

### PR DESCRIPTION
**Pull request by funbringer**

### CODE REVIEW

File: proxy/src/compute.rs [hunk 1]
```diff
@@ -110,7 +112,38 @@ pub struct PostgresConnection {
...
+                    self.config.replication_mode(ReplicationMode::Physical);
+                }
+                "database" => {
+                    self.config.replication_mode(ReplicationMode::Logical);
+                }
+                _other => {}
+            }
+        }
+
         let (socket_addr, mut stream) = self
```
Comment (hlinnaka) [inline proxy/src/compute.rs:147]: In principle, the startup packet could contain arbitrary GUCs directly as parameters. Per the PostgreSQL docs, on the [StartupMessage format](https://www.postgresql.org/docs/current/protocol-message-formats.html):

> In addition to the above, other parameters may be listed. Parameter names beginning with `_pq_.` are reserved for use as protocol extensions, while others are treated as run-time parameters to be set at backend start time. Such settings will be applied during backend start (after parsing the command-line arguments if any) and will act as session defaults.

Not sure which drivers support that in practice. I see that the tokio-postgres doesn't allow passing arbitrary params to the StartupMessage that it constructs, so passing through others than the ones listed here would need some patching of tokio-postgres. Let's at least add a comment on that.

File: test_runner/batch_others/test_proxy.py [hunk 2]
```diff
@@ -137,9 +137,8 @@ async def test_psql_session_id(vanilla_pg: VanillaPostgres, link_proxy: NeonProx
 #
 # Currently, proxy eats the extra connection options, so this fails.
 # See https://github.com/neondatabase/neon/issues/1287
-@pytest.mark.xfail
 def test_proxy_options(static_proxy):
-    with static_proxy.connect(options="-cproxytest.option=value") as conn:
+    with static_proxy.connect(options="project=foo -cproxytest.option=value") as conn:
```
[the remark this answers is not in the document]
Comment (funbringer) [reply test_runner/batch_others/test_proxy.py:141]: The name doesn't matter, we just have to provide some in the absence of SNI.

File: libs/utils/src/pq_proto.rs [hunk 3]
```diff
@@ -248,46 +264,43 @@ impl FeStartupPacket {
...
-                                let nameval: Vec<&str> = cmdopt.split('=').collect();
-                                if nameval.len() == 2 {
-                                    params.insert(nameval[0].to_string(), nameval[1].to_string());
-                                }
-                            }
-                        } else {
-                            params.insert(name.to_string(), value.to_string());
+                    // Parse pairs of null-terminated strings (key, value).
+                    let mut tokens = str::from_utf8(&params_bytes)
+                        .context("StartupMessage params: invalid utf-8")?
```
[the remark this answers is not in the document]
Comment (funbringer) [reply libs/utils/src/pq_proto.rs:330]: I made some improvements to this code, but I didn't mean to fix all of the mistakes at a time. That would be a bit too much.

File: proxy/src/compute.rs (hunk 1 above)
Comment (funbringer) [reply proxy/src/compute.rs:147]: Thank you, I'm aware of that! As you've mentioned, this requires more changes to the underlying driver. I enabled arguably the most important ones, so let's take one step at I time. Meanwhile, I've added a TODO.

**Review by kelvich — APPROVED** [review]
Thank you! Looks good, I think overall readability improved.

**Review by LizardWizzard — APPROVED** [review]
LGTM

**Pull request merged**